### ISSUE TITLE
boost defaults and conda-forge packages are incompatible (C++11)

### ISSUE THREAD

**Issue by benmwebb** [ContinuumIO/anaconda-issues#9287]
The 'boost' package exists in both defaults and conda-forge, but it appears as if the package in conda-forge was built with an older version of gcc, because it doesn't contain C++11 symbols, while the defaults package was built with a newer gcc and *does* support C++11. This is a problem if a package is built against the defaults boost and then installed on a system with conda-forge boost, or vice versa (anything that tries to use `std::string` will fail with errors like `undefined symbol: _ZNK5boost15program_options22error_with_option_name23substitute_placeholdersERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE` as it has different type in the C++11 and pre-C++11 ABIs). See salilab/conda-recipes#4 for an example.

Pinning the package to a specific version of boost doesn't help because the defaults and conda-forge packages have both the same name and version. My workaround (https://github.com/salilab/conda-recipes/tree/master/dssp) has been to put a runtime dependency on `libboost`, which doesn't appear to exist in conda-forge, thus forcing the use of the defaults package. But is there a more elegant solution?

**Comment by mingwandroid** [comment]
No, binary compatibility is the only option here. Until then you need to not mix packages in the same environment from both defaults and conda-forge.
